test/test_torch.py

```diff
@@ -2343,6 +2343,23 @@ def make_prob_dist(shape, is_contiguous):
     def test_multinomial(self):
         self._test_multinomial(self, torch.FloatTensor)
 
+    def test_multinomial_invalid_probs(self):
```

Comment:
I haven't figured out a good way to test the failure cases on CUDA - after getting `cuda runtime error (59) : device-side assert triggered` for these tests (as expected), all subsequent CUDA unit tests will fail with `RuntimeError: Creating MTGP constants failed.`, and not sure what's the right way to recover from the previous runtime error.

Reply:
afaik device-side asserts aren't recoverable. I'm wondering if you can spawn a process via multiprocessing and then trigger the assert in that process though...

Reply:
The following appears to work
```
import torch
import torch.nn as nn
import torch.multiprocessing as mp

def run():
    x = torch.randn(1, 3).cuda()
    t = torch.tensor([7]).cuda()
    model = nn.NLLLoss().cuda()
    out = model(x, t) # device assert triggered
    print(out)

if __name__ == '__main__':
    mp.set_start_method('spawn')
    p = mp.Process(target=run)
    p.start()
    p.join()

    x = torch.randn(3, 3).cuda()
    y = torch.randn(3, 3).cuda()
    print(x + y) # works!!
```

Reply:
@zou3519 Thanks for the suggestion! Works well for me too